src/lib.rs

```diff
@@ -197,20 +277,40 @@ impl KVMi {
         Ok(vcpu_count)
     }
 
-    pub fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, kvm_msrs), Error> {
+    pub fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, KvmMsr), Error> {
         let mut regs: kvm_regs = unsafe { mem::MaybeUninit::<kvm_regs>::zeroed().assume_init() };
         let mut sregs: kvm_sregs = unsafe { mem::MaybeUninit::<kvm_sregs>::zeroed().assume_init() };
-        let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        let mut msrs: KvmMsr = unsafe { mem::MaybeUninit::<KvmMsr>::zeroed().assume_init() };
         let mut mode: c_uint = 0;
+        msrs.msrs.nmsrs = 6;
+        msrs.entries[0].index = KVMiMsr::SysenterCs as u32;
+        msrs.entries[1].index = KVMiMsr::SysenterEsp as u32;
+        msrs.entries[2].index = KVMiMsr::SysenterEip as u32;
+        msrs.entries[3].index = KVMiMsr::MsrEfer as u32;
+        msrs.entries[4].index = KVMiMsr::MsrStar as u32;
+        msrs.entries[5].index = KVMiMsr::MsrLstar as u32;
         let res = (self.libkvmi.get_registers)(
-            self.dom, vcpu, &mut regs, &mut sregs, &mut msrs, &mut mode,
+            self.dom,
+            vcpu,
+            &mut regs,
+            &mut sregs,
+            &mut msrs.msrs,
+            &mut mode,
         );
         if res != 0 {
             return Err(Error::last_os_error());
         }
         Ok((regs, sregs, msrs))
     }
 
+    pub fn set_registers(&self, vcpu: u16, regs: &kvm_regs) -> Result<(), Error> {
```

Comment:
Actually, the problem that we face when implementing https://github.com/Wenzel/libmicrovmi/pull/79/files#diff-ed6d1a9ce6a167171bc40a915157b89fR140 is because we lack some abstractions from the kvmi crate.

I missed this in my review, but this public API is exposing the type `kvm_regs` from kvmi_sys.
As you defined a `pub type KVMiRegs = kvm_regs;` for this, you should reuse this type definition in this function signature too to be consistent.